proto/api.proto

```diff
@@ -376,3 +322,274 @@ message AdminMergeRequest {
+  optional Transaction pushee_txn = 2 [(gogoproto.nullable) = false];
+  // Now holds the timestamp used to compare the last heartbeat of the pushee
+  // against. This is necessary since the request header's timestamp does not
+  // necessarily advance with the node clock across retries and hence cannot
+  // detect abandoned transactions.
+  optional Timestamp now = 3 [(gogoproto.nullable) = false];
+  // Readers set this to PUSH_TIMESTAMP to move PusheeTxn's commit
+  // timestamp forward. Writers set this to ABORT_TXN to request that
+  // the PushTxn be aborted if possible. This is done in the event of
+  // a writer conflicting with PusheeTxn. Inconsistent readers set
+  // this to CLEANUP_TXN to determine whether dangling intents
+  // may be resolved.
+  optional PushTxnType push_type = 4 [(gogoproto.nullable) = false];
+  // Range lookup indicates whether we're pushing a txn because of an
+  // intent encountered while servicing an internal range lookup
+  // request. See notes in RangeLookupRequest.
+  optional bool range_lookup = 5 [(gogoproto.nullable) = false];
+}
+
+// An PushTxnResponse is the return value from the
+// PushTxn() method. It returns success and the resulting
+// state of PusheeTxn if the conflict was resolved in favor of the
+// caller; the caller should subsequently invoke
+// ResolveIntent() on the conflicted key. It returns an error
+// otherwise.
+message PushTxnResponse {
+  optional ResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+  // pushee_txn is non-nil if the transaction was pushed and contains
+  // the current value of the transaction.
+  optional Transaction pushee_txn = 2;
+}
+
+// An ResolveIntentRequest is arguments to the
+// ResolveIntent() method. It is sent by transaction
+// coordinators and after success calling PushTxn to clean up
+// write intents: either to remove them or commit them.
+message ResolveIntentRequest {
+  optional RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// An ResolveIntentResponse is the return value from the
+// ResolveIntent() method.
+message ResolveIntentResponse {
+  optional ResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// An ResolveIntentRangeRequest is arguments to the
+// ResolveIntentRange() method. This clear write intents
+// for a range of keys to resolve intents created by range ops.
+message ResolveIntentRangeRequest {
+  optional RequestHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// An ResolveIntentRangeResponse is the return value from the
+// ResolveIntent() method.
+message ResolveIntentRangeResponse {
+  optional ResponseHeader header = 1 [(gogoproto.nullable) = false, (gogoproto.embed) = true];
+}
+
+// An MergeRequest contains arguments to the Merge() method. It
+// specifies a key and a value which should be merged into the existing value at
```

Comment:
line wrapping in these comments is all over the place. would be nice to normalize it while you're here